### PR TITLE
fixes #11336 - properly report non-validation errors on product creation

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/new/product-form.controller.js [hunk 1]
```diff
@@ -38,8 +38,13 @@ angular.module('Bastion.products').controller('ProductFormController',
         function error(response) {
             $scope.working = false;
             angular.forEach(response.data.errors, function (errors, field) {
-                $scope.productForm[field].$setValidity('server', false);
-                $scope.productForm[field].$error.messages = errors;
+                if( $scope.productForm[field]) {
+                    $scope.productForm[field].$setValidity('server', false);
+                    $scope.productForm[field].$error.messages = errors;
+                }
+                else {
```
Comment (ehelms) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/new/product-form.controller.js]: `} else {`

**Comment by ehelms** [comment]
Does this actually stop the button from spinning and being greyed out?

**Comment by jlsherrill** [comment]
@ehelms it does.  It all worked fine even without this change, the error just was not displayed.

**Comment by ehelms** [comment]
ACK

**Pull request merged**